packages/.vitepress/plugins/markdownTransform.ts

```diff
@@ -264,6 +252,51 @@ function replaceAsync(str: string, match: RegExp, replacer: (substring: string,
   return Promise.all(promises).then(replacements => str.replace(match, () => replacements.shift()!))
 }
 
+/**
+ * Transpiles a TypeScript snippet to JavaScript .
+ *
+ * @param snippet The TypeScript code snippet to transpile.
+ * @returns The transpiled JavaScript code or null if no changes were made.
+ */
+async function maybeTranspileJs(snippet: string): Promise<string | null> {
+  const formattedTS = (await format(snippet.replace(/\n+/g, '\n'), { semi: false, singleQuote: true, parser: 'typescript' })).trim()
+  const js = ts.transpileModule(formattedTS, {
+    compilerOptions: { target: 99 },
+  })
+  const formattedJS = (await format(js.outputText, { semi: false, singleQuote: true, parser: 'typescript' })).trim()
+  return formattedTS === formattedJS ? null : formattedJS
+}
+
+function removeTwoslashNotations(snippet: string): string {
+  function cutAboveMarker(input: string): string {
```

Comment:
nit: can these functions be moved outside of `removeTwoslashNotations` since they can just be pure operations?

and the two `marker` consts can be outside too (like `MARKER_CUT` and `MARKER_CUT_AFTER` or some such thing)

Reply:
Just did some refactoring, readability and performance should be improved. 😃